skimage/segmentation/graph/rag.py

```diff
@@ -455,13 +455,11 @@ def rag_boundary(labels, edge_map, connectivity=2):
     data_coo = sparse.coo_matrix((data, (labels_small, labels_large)))
     graph_matrix = data_coo.tocsr()
     graph_matrix.data /= count_matrix.data
-
     rag = RAG()
     rag.add_weighted_edges_from(_edge_generator_from_csr(graph_matrix),
                                 weight='weight')
-    rag.add_weighted_edges_from(_edge_generator_from_csr(count_matrix),
```

Comment:
here, calling `add_edges_from` was overwriting the `weight` key of edges. I'm not sure the code should be modified here or in the RAG  `add_edges_from` but this fixes one of the two test failures. @jni